recipes/aaf/all/patches/0008-define-interface-for-windows.patch

```diff
@@ -0,0 +1,22 @@
+commit 2909f77ad421951cb3abf49da9af9b6317691c1a
+Author: Martin Delille <martin@delille.org>
+Date:   Tue Sep 7 16:21:39 2021 +0200
+
+    Define interface for Windows
+
+diff --git a/ref-impl/include/ref-api/AAF.h b/ref-impl/include/ref-api/AAF.h
+index 058e325cc..6d93d0f97 100644
+--- a/ref-impl/include/ref-api/AAF.h
++++ b/ref-impl/include/ref-api/AAF.h
+@@ -49,6 +49,11 @@
+ #endif
+ 
+ #ifdef __cplusplus
++
++#if defined (_WINDOWS_) && !defined (interface)
```

Comment:
What is `_WINDOWS_`? Is it defined in some headers of `aaf`?

Reply:
Ha! I think AAF SDK defines `OS_WINDOWS` in fact (based on `_WIN32`).

https://sourceforge.net/p/aaf/code2/ci/master/tree/ref-impl/include/AAFPlatform.h

Reply:
yes, `_WINDOWS_` doesn't exist in source code.

Moreover, `interface` is defined in `AAFCOMPlatformTypes.h`, so if there is some issue when `OS_WINDOWS` is defined, it should be fixed in this file.

Reply:
`_WINDOWS_` is defined in `Windows.h` (part of Windows SDK):
it's basically an include guard for that file.
if you're looking at pre-defined macro by the compiler (which is available regardless of Windows SDK headers inclusion), you're probably looking for `_WIN32`.

Reply:
`interface` is only defined in [AAFCOMPlatformTypes.h](https://sourceforge.net/p/aaf/code2/ci/master/tree/ref-impl/include/AAFCOMPlatformTypes.h#l121) if `OS_UNIX` (and not `AAF_OMIT_WINDOWS_DATA_TYPES_DEFINITIONS`). On Windows, the `#define interface struct` should come from Windows SDK headers, I thought?

Reply:
I apply the patch to `AAFCOMPlatformTypes.h`.